src/esthri/src/ops/sync.rs

```diff
@@ -483,38 +534,57 @@ where
                 bucket.clone(),
                 key.clone(),
                 directory.clone(),
+                decompress,
                 entry.unwrap(),
             )
         })
-        .map(|(s3, bucket, key, directory, entry)| async move {
-            let (path, local_etag, metadata) = entry;
-            let metadata = metadata.unwrap();
-            match local_etag {
-                Ok(local_etag) => {
-                    if local_etag != metadata.e_tag {
-                        debug!(
-                            "etag mismatch: {}, local etag={}, remote etag={}",
-                            path.display(),
-                            local_etag,
-                            metadata.e_tag
-                        );
-                        download_with_dir(&s3, &bucket, &key, &metadata.s3_suffix, &directory)
+        .map(
+            |(s3, bucket, key, directory, decompress, entry)| async move {
+                let (path, local_etag, metadata) = entry;
+                let metadata = metadata.unwrap();
```

Comment:
I see that this was more or less copy paste from the original work but seeing as how it is being touched, might also want to use an Err here.

Reply:
Yeah, I think this line is only highlighted in the diff because of indentation changes from `cargo format`. I didn't write the original code but the unwrap kind of makes sense to me here because we don't expect that the metadata will ever be none in this part of the code as it's always set in `flattened_object_listing`.

I guess it might be slightly nicer if the input stream here took `ListingMetadata` not as an option to signify this but then there'd be complexities sharing other parts of the code (particularly `map_paths_to_etags` and `local_to_remote_sync_tasks`) to handle it. The upload/download paths are probably getting distinct enough that maybe it's worth doing a bit of a refactor here to make that nicer, but I'll leave that for a different PR.